presto-product-tests/src/main/java/com/facebook/presto/tests/jdbc/JdbcTests.java

```diff
@@ -125,16 +131,22 @@ public void shouldExecuteQueryWithSelectedCatalogAndSchema()
     public void shouldSetTimezone()
             throws SQLException
     {
-        if (usingPrestoJdbcDriver(connection)) {
-            String timeZoneId = "Indian/Kerguelen";
-            ((PrestoConnection) connection).setTimeZoneId(timeZoneId);
-            try (Statement statement = connection.createStatement()) {
-                QueryResult result = queryResult(statement, "select current_timezone()");
-                assertThat(result).contains(row(timeZoneId));
-            }
+        String timeZoneId = "Indian/Kerguelen";
+        Connection testConnection = connection;
+        if (usingPrestoJdbcDriver(testConnection)) {
+            ((PrestoConnection) testConnection).setTimeZoneId(timeZoneId);
+        }
+        else if (usingTeradataJdbcDriver(testConnection)){
+            testConnection = DriverManager.getConnection(
+                    String.format("jdbc:presto://%s:%d;TimeZoneID=%s", prestoServer, prestoPort, timeZoneId));
         }
         else {
-            LOGGER.warn("shouldSetTimezone() only applies to PrestoJdbcDriver");
+            throw new IllegalStateException();
```

Comment:
Add a message -- something like "JDBC driver must either be the Presto open source driver or the Teradata driver"